DevelopersGuide.adoc

```diff
@@ -0,0 +1,11425 @@
+
+=== Setup for Playing with Loads
+
+It is very handy to be able to look at your application's state to see what might be wrong. We've been manually
+dumping application state at the REPL using a rather long expression. So, at this point make sure
+you are either running your application in a devcard, or you know how to look at things with Fulcro
+Inspect. The output in the devcards is typically easier for beginners to read.
+
+=== Loading Data [[Loading]]
+
+Now we will start to see more of the payoff of our UI co-located queries and auto-normalization. Our application
+so far is quite unrealistic: the people we're showing should be coming from a server-side database, they
+should not be embedded in the code of the client. Let's remedy that.
+
+Fulcro provides a few mechanisms for loading data, but every possible load scenario can be done using
+the `fulcro.client.data-fetch/load` function.
+
+It is very important to remember that our application database is completely normalized, so anything we'd want to put
+in that application state will be *at most* 3 levels deep (the table name, the ID of the thing in the table, and the
+field within that thing). We've also seen that Fulcro can also auto-normalize complete trees of data,
+and has graph queries that can be used to ask for those trees.
+
+Thus, there really are not very many scenarios!
+
+The three basic scenarios are:
+
+* Load something into the root of the application state
+* Load something into a particular field of an existing thing
+* Load some pile of data, and shape it into the database (e.g. load all of the people, and then separate them into
+a list of friends and enemies).
+
+Let's try out these different scenarios with our application.
+
+First, let's correct our application's initial state so that no people are there:
+
+```
+(defsc PersonList [this {:keys [db/id person-list/label person-list/people]}]
+  {:query [:db/id :person-list/label {:person-list/people (prim/get-query Person)}]
+   :ident [:person-list/by-id :db/id]
+   :initial-state
+          (fn [{:keys [id label]}]
+            {:db/id              id
+             :person-list/label  label
+             :person-list/people []})} ; REMOVE THE INITIAL PEOPLE
+  ...
+```
+
+If you now reload your page you should see two empty lists.
+
+==== Normalization
+
+When you load something you will use a query from something on your UI (it is rare to load something you don't want to
+show). Since those components (should) have a query and ident, the result of a load can be sent from the server as a
+tree, and the client can auto-normalize that tree just like it did for our initial state!
+
+==== Loading something into the DB root
+
+This case is less common, but it is a simple starting point. It is typically used to obtain something that you'd want
+to access globally (e.g. the user info about the current session). Let's assume that our Person component represents
+the same kind of data as the "logged in" user. Let's write a
```

Comment:
This representation does not show the data structure above it. Joe should point to Heather and vice versa. Julie shouldn't have a spouse. Julie instead of Heather should be a child of Joe.

Reply:
So, github isn't showing your comments in the right place. If you have any more, copy a sentence from the actual doc and include it for reference so I can easily find where you're looking...you could also just copy the section heading wih like "==== Idents"...just something distinct I can find.

This is fixed and I did another forced push on book branch to get some other patches in from develop.